man/fi_av.3.md

```diff
@@ -450,6 +485,51 @@ size of the buffer needed to store the address.  This size may be
 larger than the input len.  If the provided buffer is too small, the
 results will be truncated.  fi_av_straddr returns a pointer to buf.
 
+## fi_av_insert_auth_key
+
+This function associates authorization keys with an address vector. This
+behavior, combined with endpoints opened with `FI_AV_AUTH_KEY``, enables an
+endpoint to support multiple authorization keys. When an endpoint with
+`FI_AV_AUTH_KEY`` is enabled, the endpoint is configured against
+authorization keys inserted at that point in time. Later authorization
+key insertions will not propagate to already enabled endpoints.
+
+The `auth_key` and `auth_key_size` parameters are used to input the
+authorization key into the address vector. The structure of the
+authorization key is provider specific.
+
+The output of `fi_av_insert_auth_key` is a fi_addr_t handle representing
+all endpoint addresses against this specific authorization key. For
+endpoints enabled with FI_DIRECTED_RECV, authorization key fi_addr_t's
+inserted with `FI_RECV` can be used to restrict incoming messages to only
+endpoint addresses within the authorization key.
+
+These authorization key fi_addr_t's can later be used an input for
+endpoint address insertion functions to generate an fi_addr_t for a
+specific endpoint address and authorization key.
+
+*flags*
+: The following flag may be passed to `fi_av_insert_auth_key`:
+
+- *FI_TRANSMIT*
+: Restrict the authorization key to outbound data transfers. This includes
+  send message, RMA, and atomic operations.
+
+- *FI_RECV*
+: Restrict the authorization key to inbound data transfers. This includes
+  received messages and target MRs of RMA and atomic operations.
```

Comment:
If the access flags aren't immediately usable, I would leave flags as reserved for now.  The issues is there's no easy way to extend flags in the future to have an auth_key for message sends that's separate from that used for RMA writes, or an auth_key for message receives from remote RMA.

Is that separation useful?  I don't know.  Honestly it seems doubtful.  But if we can avoid limiting the use of flags until we have a use case, it would be better.

Reply:
Sounds good. So the default behavior would be to enable tx/rx on the authorization key.